### PR TITLE
Re-implemented setName for tools

### PR DESCRIPTION
**Pull request by fmauch**
In the migration towards RViz2 dynamically setting a tool name seems to got dropped. This PR adds this feature again.

We use one central RViz instance in a multi-mobile-robot application where we have a NavGoal tool for each robot. Not being able to rename the tools is a bit hard to use as you'll have to memorize the actual order.

I don't know whether there is a specific reason why this got dropped, but I think this is valid and useful to keep.

As we currently use ROS 2 Humble, I think this should also be backported to Humble. I can open a second PR if desired.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
uncrustify https://build.ros2.org/job/Rpr__rviz__ubuntu_jammy_amd64/177/

**Comment by fmauch** [comment]
@ahcorde should be fixed now.

**Comment by ahcorde** [comment]
@fmauch, can you share a gif or explain how this feature work?

**Comment by fmauch** [comment]
Sorry for being so brief, I was assuming that this got kicked out when migrating to RViz2 so I didn't add much explanation. Reading back, it might be that the changes were added to RViz1 in December 2020 after RViz2 was forked out.

Basically, this goes back to [this issue](https://github.com/ros-visualization/rviz/issues/1138) and [this PR](https://github.com/ros-visualization/rviz/pull/1570) for ROS1 RViz. Reasoning for the changes is well explained in https://github.com/ros-visualization/rviz/pull/1570#issuecomment-747450011.

TLDR: calls to `setName()` after the tool has been created doesn't have any effect in RViz 2.

I've created a custom tool with a `name` property, so that we can set its name manually, similar to the action namespace
of the goal tool. Changing the property works fine and it gets saved to the RViz configuration, but it is never applied to the actual tool. The following screenshot shows the tool (the one with the green flag) with its properties opened below. As you can see, the name got applied correctly, the tool also shows up under the new name in the properties view. However, in the toolbar the tool still has the default name "2D Nav Goal".

image

Basically, our initialization looks as follows:
```c++
void NavGoalTool::onInitialize()
{
  PoseTool::onInitialize();
  setName(m_name_ptr->getString());
  
// ... more unrelated stuff here...
}
```

where `m_name_ptr` has been initialized with

```c++
  m_name_ptr.reset(new rviz_common::properties::StringProperty(
    "Name", "2D Nav Goal", "The display name", getPropertyContainer(), SLOT(updateName()), this));
```

The `updateName()` method looks like this:

```c++
void NavGoalTool::updateName()
{
  setName(m_name_ptr->getString());
}
```

The [docs of `setName()`](https://github.com/ros2/rviz/blob/75322dccc077a219a9e432489af763243e8857f1/rviz_common/include/rviz_common/tool.hpp#LL129-L136) suggest that the name can be altered lateron, as well which is not completely wrong, as the actual tool gets renamed in the properties view.


With the changes from this PR applied, the same screenshot with an RViz2 started from the same config looks like this:

image

With this modification we can at least copy the tools inside the config file and rename them so we get a goal tool for each robot:
image

**Review by clalancette — APPROVED** [review]
Thanks for the detailed explanation @fmauch .  In short, this looks good to me, so I'll run CI on it next.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18973/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13500/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/19719/)

**Review by ahcorde — APPROVED** [review]

**Comment by fmauch** [comment]
Thank you for merging. Is there any chance this could get backported to Humble?

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport iron humble
